File: sdk/python/legend_ai/resources/market.py
"""
Market data resource
"""

from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from ..client import LegendAI, AsyncLegendAI


class MarketResource:
    """Synchronous market data resource"""

    def __init__(self, client: "LegendAI"):
        self._client = client

    def internals(self) -> dict:
        """Get market internals (breadth, advance/decline, regime)"""
        return self._client.request("GET", "/api/market/internals")

    def breadth(self) -> dict:
        """Get market breadth metrics"""
        return self._client.request("GET", "/api/market/breadth")

    def regime(self) -> dict:
        """Get market regime classification"""
        return self._client.request("GET", "/api/market/regime")


class AsyncMarketResource:
    """Asynchronous market data resource"""

    def __init__(self, client: "AsyncLegendAI"):
        self._client = client

    async def internals(self) -> dict:
        """Get market internals (async)"""
        return await self._client.request("GET", "/api/market/internals")

    async def breadth(self) -> dict:
        """Get market breadth metrics (async)"""
        return await self._client.request("GET", "/api/market/breadth")

    async def regime(self) -> dict:
        """Get market regime classification (async)"""
        return await self._client.request("GET", "/api/market/regime")

File: sdk/javascript/src/resources/market.ts
/**
 * Market data resource
 */

import type { AxiosInstance } from 'axios';
import type { MarketInternalsResponse } from '../types';

export class MarketResource {
  constructor(private client: AxiosInstance) {}

  /**
   * Get market internals (breadth, advance/decline, regime)
   */
  async internals(): Promise<MarketInternalsResponse> {
    const response = await this.client.get<MarketInternalsResponse>('/api/market/internals');
    return response.data;
  }

  /**
   * Get market breadth metrics
   */
  async breadth(): Promise<Record<string, any>> {
    const response = await this.client.get('/api/market/breadth');
    return response.data;
  }

  /**
   * Get market regime classification
   */
  async regime(): Promise<Record<string, any>> {
    const response = await this.client.get('/api/market/regime');
    return response.data;
  }
}
